clog/readers.py

```diff
@@ -406,6 +405,19 @@ def list_streams(self):
         towrite = ""
         return NetCLogStreamReader._ContextManager(self, towrite)
 
+def construct_conn_msg(stream, lines=None, protocol_opts=None):
+    """Return a connnection message
+
+    :param stream: stream name
+    :param lines: number of messages to consume
+    :param protocol_opts: optional arguments
+    """
+    connection_msg = stream
+    if lines:
+        connection_msg += ' {0}'.format(lines)
+    if protocol_opts:
+        connection_msg += ''.join([' {0}={1}'.format(k, v) for k, v in protocol_opts.items()])
```

Comment:
it's very unlikely to make a performance difference here (especially with a very small dictionary).  It may actually make it slower due to iterator overhead